api/envoy/extensions/filters/http/custom_response/v3/custom_response.proto

```diff
@@ -0,0 +1,86 @@
+syntax = "proto3";
+
+package envoy.extensions.filters.http.custom_response.v3;
+
+import "envoy/config/core/v3/base.proto";
+import "envoy/config/core/v3/substitution_format_string.proto";
+
+import "google/protobuf/wrappers.proto";
+
+import "xds/annotations/v3/status.proto";
+import "xds/type/matcher/v3/matcher.proto";
+
+import "udpa/annotations/status.proto";
+import "validate/validate.proto";
+
+option java_package = "io.envoyproxy.envoy.extensions.filters.http.custom_response.v3";
+option java_outer_classname = "CustomResponseProto";
+option java_multiple_files = true;
+option go_package = "github.com/envoyproxy/go-control-plane/envoy/extensions/filters/http/custom_response/v3;custom_responsev3";
+option (udpa.annotations.file_status).package_version_status = ACTIVE;
+option (xds.annotations.v3.file_status).work_in_progress = true;
+
+// [#protodoc-title: Custom Response Filter]
+// [#extension: envoy.filters.http.custom_response]
+
+// The Custom Response Filter allows for replacing upstream responses.
+
+// The filter configuration is a collection of custom response
+// policies.
+message CustomResponse {
+  // Definition of the custom response to be provided to the downstream.
+  // [#next-free-field: 6]
+  message Response {
+    message RemoteDataSource {
+      // The Http URI to fetch the custom response from.
+      // It should be a full FQDN with protocol, host and path.
+      //
+      // Example:
+      //
+      // .. code-block:: yaml
+      //
+      //    uri: https://www.googleapis.com/oauth2/v1/certs
```

Comment:
Certs is probable not the best example.
Can this be an actual example of contents sent as a custom response?
If not, just state `https://www.mydomain.com/path/to/404.txt` or something